### ISSUE TITLE
Error when repeating request to play a playlist 

### ISSUE THREAD

**Issue by richardvolpe** [jishi/node-sonos-http-api#625]
Hi 

I have built a python script that reads RFID cards and then based on the UID of the card, plays a particular Sonos playlist (via Spotify) using the following line of code:

`urllib2.urlopen("http://localhost:5005/Kitchen/playlist/name_of_playlist").read()`

The script works working perfectly, for the first RFID tap, and the second RFID tap, but strangely when I attempt the THIRD tap it returns the following error:

File "Sonos.py", line 120, in <module>
    urllib2.urlopen("http://localhost:5005/Kitchen/playlist/name_of_playlist").read()
  File "/usr/lib/python2.7/urllib2.py", line 154, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python2.7/urllib2.py", line 435, in open
    response = meth(req, response)
  File "/usr/lib/python2.7/urllib2.py", line 548, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib/python2.7/urllib2.py", line 473, in error
    return self._call_chain(*args)
  File "/usr/lib/python2.7/urllib2.py", line 407, in _call_chain
    result = func(*args)
  File "/usr/lib/python2.7/urllib2.py", line 556, in http_error_default
    raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
urllib2.HTTPError: HTTP Error 500: Internal Server Error

In attempting to debug, I have circumvented all the RFID side of things and simply used a browser to browse to "http://localhost:5005/Kitchen/playlist/name_of_playlist" and I am getting the same weird outcome - i.e  the first 2 times I browse to that page, I get successful playing of the playlist, and the 3rd time I get this error message:

{"status":"error","error":"connect EHOSTUNREACH 192.168.0.19:1400","stack":"Error: connect EHOSTUNREACH 192.168.0.19:1400\n    at Object._errnoException (util.js:992:11)\n    at _exceptionWithHostPort (util.js:1014:20)\n    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1186:14)"}

It seems like I am overloading the Sonos every third attempt to send it an instruction?

Lastly I should mention that I tried using the API to access the Sonos simply to pull information (rather than instruct it to do something, such as playing a playlist) and I have no issues, no matter how many times I perform the following request: 

http://localhost:5005/zones

Would really appreciate any assistance!!

Thanks
Richard

**Comment by jishi** [comment]
That is weird, it seems like the connectivity to the player you are sending requests to gets lost, like the player is changing IP (but that would cause you other problems with the system, like lost playlists or stopped playback etc). 

How are you running the http api? How is it connected to your network? How are your players connected to the network?

**Comment by richardvolpe** [comment]
Hi Jimmy

I am running the api on a raspberry pi 3 running raspbian and connected
with wifi. The sonos system too is using wifi (regular, not their
proprietary wireless network).

The sonos system is otherwise operating normally...

**Comment by jishi** [comment]
Could you try, when you get that error in the log (EHOSTUNREACH), to try and open http://192.168.0.19:1400/status or whatever IP it is reporting in the error, from another computer, and see if that works?

**Comment by richardvolpe** [comment]
Thanks, I tried that and strangely there was no Sonos unit on that IP (i.e. it just went to a "host unreachable" page).
On further investigation, I think I have uncovered the problem... I have 3 Sonos devices, 2 which connect to my primary router and 1 which connects to a secondary router which is providing an extension to my wifi network. When I switched off the Sonos device which was connected to the secondary router, everything is now working perfectly. I think there was some issue relating to Sonos's inter-speaker communication being broken by the 2 router setup. Very strange. 

Bottom line, I am guessing this is an issue with my network configuration. At least now I know what I need to fix (and its not your API software!)

Thanks again for your help.

**Comment by jishi** [comment]
No problem. The secondary router is hopefully configured in access point mode, otherwise it would probably not have been found in the first place. It might get conflicting IPs though if the secondary router is misconfigured (aka, DHCP is not disabled). Glad you found the root cause :)
